mmf/modules/encoders.py

```diff
@@ -338,6 +341,47 @@ def forward(self, x):
         return x["res5"]
 
 
+@registry.register_encoder("frcnn")
+class FRCNNImageEncoder(Encoder):
+    @dataclass
+    class Config(Encoder.Config):
+        name: str = "frcnn"
+        pretrained: bool = True
+        pretrained_path: str = None
+
+    def __init__(self, config: Config, *args, **kwargs):
+        super().__init__()
+        self.config = config
+        # pretrained = config.get("pretrained", False)
+        # pretrained_path = config.get("pretrained_path", None)
+
+        if config.pretrained:
+            self.frcnn = GeneralizedRCNN.from_pretrained(
+                config.pretrained_path, config=config
+            )
```

Comment:
Let's get rid of the from_pretrained method. You can directly download the checkpoint and then use load_state_dict to load the checkpoint after the model is initialized from config. 

Check how it is done in `Detectron2ResnetImageEncoder` above.

Reply:
Might be useful to keep it as well for future use cases and then use load_state_dict here.